source/extensions/quic/connection_debug_visitor/quic_stats/quic_stats.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/extensions/quic/connection_debug_visitor/quic_stats/v3/quic_stats.pb.h"
+#include "envoy/registry/registry.h"
+
+#include "source/common/protobuf/protobuf.h"
+#include "source/common/quic/envoy_quic_connection_debug_visitor_factory_interface.h"
+
+#include "quiche/quic/core/frames/quic_connection_close_frame.h"
+#include "quiche/quic/core/quic_connection.h"
+#include "quiche/quic/core/quic_session.h"
+#include "quiche/quic/core/quic_types.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Quic {
+namespace ConnectionDebugVisitors {
+namespace QuicStats {
+
+#define ALL_QUIC_STATS(COUNTER, GAUGE, HISTOGRAM)                                                  \
+  COUNTER(cx_tx_packets_total)                                                                     \
+  COUNTER(cx_packets_retransmitted_total)                                                          \
+  COUNTER(cx_amplification_throttling_total)                                                       \
+  COUNTER(cx_path_degrading_total)                                                                 \
+  COUNTER(cx_forward_progress_after_path_degrading_total)                                          \
+  HISTOGRAM(cx_tx_percent_retransmitted_pkts, Percent)                                             \
+  HISTOGRAM(cx_srtt_us, Microseconds)                                                              \
+  HISTOGRAM(cx_tx_estimated_bandwidth_bytes_per_second, Bytes)
+
+struct QuicStats {
+  ALL_QUIC_STATS(GENERATE_COUNTER_STRUCT, GENERATE_GAUGE_STRUCT, GENERATE_HISTOGRAM_STRUCT)
+};
+
+class Config;
+
+// Visitor class that writes connection-level information to info logs.
```

Comment:
This might be a dumb question but does this write to info logs, or to counters?

Reply:
Oh, that's an incorrect copy/paste. I started with the `basic` visitor. This extension writes to counters/gauges/histograms, NOT to logs.